animation-glib/glide/glide.cpp

```diff
@@ -0,0 +1,404 @@
+
+  return animation_stepper_wrapper_new ((gpointer) &stepper);
+}
+
+static void
+animation_glide_animation_set_property (GObject      *object,
+                                        guint         prop_id,
+                                        const GValue *value,
+                                        GParamSpec   *pspec)
+{
+  AnimationGlideAnimation *glide_animation = ANIMATION_GLIDE_ANIMATION (object);
+
+  switch (prop_id)
+    {
+    case PROP_INITIAL_DISTANCE:
+      animation_glide_animation_set_initial_distance (glide_animation, g_value_get_float (value));
+      break;
+    case PROP_X_ROTATION_ANGLE_DEGREES:
+      animation_glide_animation_set_x_rotation_angle_degrees (glide_animation, g_value_get_float (value));
+      break;
+    case PROP_Y_ROTATION_ANGLE_DEGREES:
+      animation_glide_animation_set_y_rotation_angle_degrees (glide_animation, g_value_get_float (value));
+      break;
+    case PROP_X_AXIS_LOCATION_UNIT:
+      animation_glide_animation_set_x_axis_location_unit (glide_animation, g_value_get_float (value));
+      break;
+    case PROP_Y_AXIS_LOCATION_UNIT:
+      animation_glide_animation_set_y_axis_location_unit (glide_animation, g_value_get_float (value));
+      break;
+    case PROP_SCREEN_WIDTH:
+      /* Not writable, except on construction */
+      break;
+    case PROP_TARGET:
+      /* Not writable, except on construction */
+      break;
+    case PROP_STEPPER:
+      animation_glide_animation_set_stepper (glide_animation,
+                                             ANIMATION_STEPPER (g_value_get_object (value)));
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+    }
+}
+
+static void
+animation_glide_animation_get_property (GObject    *object,
+                                        guint       prop_id,
+                                        GValue     *value,
+                                        GParamSpec *pspec)
+{
+  AnimationGlideAnimation *glide_animation = ANIMATION_GLIDE_ANIMATION (object);
+
+  switch (prop_id)
+    {
+    case PROP_INITIAL_DISTANCE:
+      g_value_set_float (value, animation_glide_animation_get_initial_distance (glide_animation));
+      break;
+    case PROP_X_ROTATION_ANGLE_DEGREES:
+      g_value_set_float (value, animation_glide_animation_get_x_rotation_angle_degrees (glide_animation));
+      break;
+    case PROP_Y_ROTATION_ANGLE_DEGREES:
+      g_value_set_float (value, animation_glide_animation_get_y_rotation_angle_degrees (glide_animation));
+      break;
+    case PROP_X_AXIS_LOCATION_UNIT:
+      g_value_set_float (value, animation_glide_animation_get_x_axis_location_unit (glide_animation));
+      break;
+    case PROP_Y_AXIS_LOCATION_UNIT:
+      g_value_set_float (value, animation_glide_animation_get_y_axis_location_unit (glide_animation));
+      break;
+    case PROP_SCREEN_WIDTH:
+      /* Not readable */
+      break;
+    case PROP_TARGET:
+      /* Not readable */
+      break;
```

Comment:
Remove these two; if a property is not readable you're guaranteed it won't appear here.